### ISSUE TITLE
improper utf-8 handling in json-unit-spring

### ISSUE THREAD

**Issue by octopus-prime** [lukas-krecan/JsonUnit#212]
**Describe the bug**

- JsonUnit 2.11.0
- BringBoot 2.2.0.RELEASE

When using `json-unit` with `rest-assured` all works fine:
```
        given(specification)
                .with()
                .accept(MediaType.APPLICATION_JSON_VALUE)
                .queryParam("produktIds", PreisApiData.PRODUKT_IDS)

                .when()
                .get("preise")

                .then()
                .statusCode(HttpStatus.OK.value())

                .assertThat()
                .contentType(MediaType.APPLICATION_JSON_VALUE)
                .body(jsonEquals(PreisApiData.PREIS_DTOS));
```

But writing the same test with `json-unit-spring` and `mock-mvc` causes encoding issues:
```
        mockMvc.perform(get("/preise")
                .accept(MediaType.APPLICATION_JSON)
                .param("produktIds", PreisApiData.PRODUKT1_ID.toString(), PreisApiData.PRODUKT2_ID.toString()))
                .andExpect(status().isOk())
                .andExpect(content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
                .andExpect(json().isEqualTo(PreisApiData.PREIS_DTOS));
```
->
```
Different value found in node "[0].currency", expected: <"€"> but was: <"â¬">.
Comparison Failure: 
Expected :€
Actual   :â¬
```

**Comment by octopus-prime** [comment]
Maybe in `JsonUnitResultMatchers` the code could be changed from
```
        @Override
        public void match(MvcResult result) throws Exception {
            String actual = result.getResponse().getContentAsString();
            doMatch(actual);
        }
```
to
```
        @Override
        public void match(MvcResult result) throws Exception {
            String actual = result.getResponse().getContentAsString(StandardCharsets.UTF_8);
            doMatch(actual);
        }

```

**Comment by lukas-krecan** [comment]
Good catch, thanks, will fix it.

**Comment by lukas-krecan** [comment]
It seems that it's a problem on your side. Spring takes the encoding form the response content type. Can you check the 'Content-Type' header returned by the controller? I was able to reproduce it like this (you should do the opposit :-))

**Comment by octopus-prime** [comment]
Note that 
`MediaType.APPLICATION_JSON_UTF8`
and
`MediaType.APPLICATION_JSON_UTF8_VALUE`
are deprecated since Spring 5.2 (and so in Spring Boot 2.2)

So the controller implements this API
```
@Tag(name = "Preis")
public interface PreisApi {

    @Operation(summary = "Get kunde")
    @GetMapping(path = "preise", produces = MediaType.APPLICATION_JSON_VALUE)
    @ResponseStatus(HttpStatus.OK)
    Set<PreisDto> getPreise(@RequestParam("produktIds") Set<UUID> produktIds);
}
```

**Comment by lukas-krecan** [comment]
Ok, it's quite a mess but this should help https://github.com/lukas-krecan/JsonUnit/pull/213

**Comment by lukas-krecan** [comment]
Released as 2.11.1
